spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -569,7 +569,25 @@ public void setAfterReceivePostProcessors(MessagePostProcessor... afterReceivePo
 	}
 
 	private void evaluateFastReplyTo() {
-		this.usingFastReplyTo = false;
+		this.usingFastReplyTo = useDirectReplyTo();
+		this.evaluatedFastReplyTo = true;
+	}
+
+	/**
+	 * Override this method use some other criteria to decide whether or not to use
+	 * direct reply-to (https://www.rabbitmq.com/direct-reply-to.html).
+	 * The default implementation returns true if the broker supports it and there
+	 * is no {@link #setReplyAddress(String) replyAddress} set and the replyAddress
+	 * is not "##TEMPORARY_QUEUE##". When direct reply-to is not used, the template
+	 * will create a temporary, auto-delete queue for the reply.
+	 * <p>
+	 * This method is invoked once only - when the first message is sent.
+	 * @return true to use direct reply-to.
+	 */
+	protected boolean useDirectReplyTo() {
```

Comment:
It is called from `evaluatedFastReplyTo` which is called from a `synchronized` block.